modules/table/src/main/java/org/apache/ignite/internal/table/distributed/replicator/PartitionReplicaListener.java

```diff
@@ -2906,18 +2906,20 @@ private CompletableFuture<CommandApplicationResult> applyUpdateAllCommand(
                     return safeTime.waitFor(safeTs)
                             .thenApply(ignored -> new CommandApplicationResult(safeTs, null));
                 } else {
+                    HybridTimestamp safeTs = ((UpdateAllCommand) res.getCommand()).safeTime();
```

Comment:
Why do we take safeTs from command and not result? And why wasn't this changed for `UpdateCommand` handling where everything seems to be the same?

Reply:
Good cache. This is incorrect, fixed.